code/modules/pda/core_apps.dm

```diff
@@ -40,27 +45,32 @@
 /datum/data/pda/app/notekeeper
 	name = "Notekeeper"
 	icon = "sticky-note-o"
-	template = "pda_notekeeper"
+	template = "pda_notes"
 
-	var/note = null
-	var/notehtml = ""
+	var/note
 
 /datum/data/pda/app/notekeeper/start()
 	. = ..()
 	if(!note)
 		note = "Congratulations, your station has chosen the [pda.model_name]!"
 
 /datum/data/pda/app/notekeeper/update_ui(mob/user as mob, list/data)
-	data["note"] = note									// current pda notes
+	data["note"] = note	// current pda notes
+
+/datum/data/pda/app/notekeeper/tgui_act(action, params)
+	if(..())
+		return
+
+	. = TRUE
 
-/datum/data/pda/app/notekeeper/Topic(href, list/href_list)
-	switch(href_list["choice"])
+	switch(action)
 		if("Edit")
-			var/n = input("Please enter message", name, notehtml) as message
+			var/n = input("Please enter message", name, note) as message
 			if(pda.loc == usr)
+				// TGUI will auto-reject supplied HTML
+				// However, the admin var-edit window will not
+				// SANITISATION IS IMPORTANT. DO NOT NEGLECT.
```

Comment:
![image](https://user-images.githubusercontent.com/25063394/94195850-755b4580-feab-11ea-8f7a-e66481bf61bf.png)

Reply:
The next line down
```dm
				// TGUI will auto-reject supplied HTML
				// However, the admin var-edit window will not
				// SANITISATION IS IMPORTANT. DO NOT NEGLECT.
				note = adminscrub(n)

//Runs byond's sanitization proc along-side strip_html_simple
//I believe strip_html_simple() is required to run first to prevent '<' from displaying as '&lt;' that html_encode() would cause
/proc/adminscrub(var/t,var/limit=MAX_MESSAGE_LEN)
	return copytext((html_encode(strip_html_simple(t))),1,limit)
```